### DISCUSSION TITLE
Add a color picker

### DISCUSSION THREAD

**Discussion by laycookie** [shadcn-ui/ui#573] [Ideas]
Recently needed a color piker component in my project and I'm just wondering if it will make sense to have one here, Shadcn/ui does have a calendar so why not a color piker? I'm interested if this component is in plans or can I make one and open a pull request?

**Comment by NikharPandya** [comment]
@laycookie , at this moment components are built with radix ui primitives. So will only have components made with them. Stay tuned in future for updates.

**Comment by Illyism** [comment]
<img width="302" alt="image" src="https://github.com/shadcn-ui/ui/assets/304283/c9b2ea02-73f0-4583-b8a3-b830cc0c80ec">


@laycookie I made a color + gradient + image picker.

Docs: https://gradient.page/picker
Github: https://github.com/illyism/gradient-picker

**Reply by FelipeCosta401** [reply]
I'm trying to use this example with Vite and i'm getting a 504 status code. I saw u using "use-client" in the begining of the code, i think thats next.js, right? Is there a way for me to use ur component with Vite ?? I'm kind lost because I still learing how to code. May someone help me pls? thx

**Reply by nathan-hello** [reply]
@FelipeCosta401 "use client" just means that it's a normal React component. There isn't anything different about NextJS client components and React components in Vite. If you're seeing a 504 in your console log, could you share what the output of your terminal is that's running vite? There should be some error information there.

**Reply by kerryj89** [reply]
Documentation link is dead.  Seems a bit spammy with the ad/commission stuff weaved into the demo.  Looks like it's good for picking premade selections, a color picker in the simplest sense with no color wheel.  Strangely there is an input to override and enter your own color?  Maybe that's an optional thing.

**Comment by PabloResende** [comment]
😎😎😎

**Reply by Illyism** [reply]
Add one more example:
- https://gradient.page/hex/CCFF00

**Reply by untitledappseu** [reply]
Yeah please add Color Picker

**Comment by nightspite** [comment]
Example: https://shadcn-color-picker.vercel.app

https://github.com/shadcn-ui/ui/assets/26671956/e107749c-2fd7-45df-b286-73e22fa5ac48

**Reply by kerryj89** [reply]
I think this is great for simple customer edits e.g. branding colors for customers.  A bit strange that there is no spacing between the picker area and input but easily fixable.

**Comment by Absleepy** [comment]
@Illyism The link you shared is great. But it's not a color picker. It's more of a color selector.
Would be great if user could drag around on a color picker. 
Cheers

**Comment by cael-prussian** [comment]
https://v0.dev/chat/color-picker-component-2sBBFCIJr4q?b=b_osv3Z2Pc2fg

**Reply by mlc-studio** [reply]
There is a bug with your color picker (try up right corner, instead of the color you have white)
![image](https://github.com/user-attachments/assets/e9855486-d8c3-4d7e-926c-35d156a14d70)

**Reply by cael-prussian** [reply]
The top edge should actually be full light/white. The issue is the styling of the HSA diagram isn't great, I slapped it together in 5 minutes without touching any code using v0 😅 so probably needs a bunch of testing before being used anywhere.

Using a design tool like Figma would probably get a better gradient for the HSA picker area but something like this works (line 150 of advanced-color-picker.tsx)  

`
                linear-gradient(
                  180deg,
                  #fff 0%,
                  rgba(128, 128, 128, 0) 50%,
                  #000 100%
                ),
                radial-gradient(
                  ellipse at 100% 50%,
                  hsl(${hsl.h}, 100%, 50%) 0%,
                  transparent 100%
                )
                `
https://v0.dev/chat/color-picker-component-2sBBFCIJr4q?b=b_osv3Z2Pc2fg&p=0

**Reply by yymm120** [reply]
very beautiful, but many bugs. 
Looking forward to it.

**Reply by kerryj89** [reply]
I like this but if you drag the hue slider to the end quickly it resets to the left.  It feels like a color picker catered towards more advanced users or WYSIWYG editors, almost like a color picker app to itself :)

**Comment by vatsalpipalava** [comment]
<img width="1523" height="705" alt="shadcn-input-color" src="https://github.com/user-attachments/assets/5c919c69-4f91-4413-bdab-4d2a2c0fffd7" />

https://shadcn-input-color.vercel.app

**Reply by kerryj89** [reply]
Best one here that I've tried, nice work

**Comment by karticme** [comment]


**Reply by kerryj89** [reply]
This one is also nice, but a bit unexpected that opacity control locks you into hex mode.

**Reply by karticme** [reply]
Yes, that's because of width issue but working on it and will be fixed and releasing it to public use shortly.

**Comment by karticme** [comment]
@shadcn I have created this Color Picker component with React Aria Components and Shadcn UI. Take a look https://kartic-colors.vercel.app
It's fully modular, customisable and clean. Let me know what do you think about it.
I want contribute with this component.
-&nbsp;[kartic](https://x.com/01_kartic)

**Comment by Prajwal-alphanimble** [comment]
you might wanna check out this discussion where I implemented a new resizable and dragable dashboard layout... this can be very helpfull to many developers in making analytic, sharable dashboard, the users can alter the dashboard based on their liking...

The disscussion link: https://github.com/shadcn-ui/ui/discussions/10466

**Comment by TheAleSch** [comment]
I have added a new compliant color picker that supports different modes, it's on the registry already and you can find a demo at https://amplo.ale.design/ - completely accessible and supporting oklch and p3 color spaces.

**Reply by brentertz** [reply]
Super nice! Any plans for a base-ui version?

**Reply by TheAleSch** [reply]
For sure! I'm just finishing the gradient fill picker first.

**Reply by TheAleSch** [reply]
> Super nice! Any plans for a base-ui version?

@brentertz baseUI live =) and the default

**Reply by Cruel** [reply]
You killed it. 👍
I'm trying it out now.
